src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,220 @@
+        }
+    }
+
+    private Car getCarFromResultSet(ResultSet resultSet) throws SQLException {
+        Car car = new Car();
+        car.setId(resultSet.getObject(CAR_ID_COLUMN, Long.class));
+        car.setModel(resultSet.getString(CAR_MODEL_COLUMN));
+        Manufacturer manufacturer = new Manufacturer();
+        manufacturer.setId(resultSet.getObject(CAR_MANUFACTURER_ID_COLUMN, Long.class));
+        manufacturer.setName(resultSet.getString(MANUFACTURER_NAME_COLUMN));
+        manufacturer.setCountry(resultSet.getString(MANUFACTURER_COUNTRY_COLUMN));
+        car.setManufacturer(manufacturer);
+        return car;
+    }
+
+    private List<Driver> getDriversForCar(Long carId) {
+        String getDriverQuery = "SELECT id, name, license_number "
+                + "FROM drivers d JOIN cars_drivers cd ON cd.driver_id = d.id "
+                + "WHERE d.is_deleted = FALSE AND cd.car_id = ?;";
+        List<Driver> drivers = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement =
+                            connection.prepareStatement(getDriverQuery)) {
+            preparedStatement.setLong(1, carId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                drivers.add(getDriverFromResultSet(resultSet));
+            }
+            return drivers;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get the driver by carId " + carId, e);
+        }
+    }
+
+    private Driver getDriverFromResultSet(ResultSet resultSet) throws SQLException {
+        return new Driver(
+                resultSet.getObject(DRIVER_ID_COLUMN, Long.class),
+                resultSet.getString(DRIVER_NAME_COLUMN),
+                resultSet.getString(DRIVER_LICENSE_NUMBER_COLUMN)
+        );
+    }
+
+    private void deleteCarFromCarsDrivers(Long carId) {
+        String deleteCarIdFromCarsDriversQuery = "DELETE FROM cars_drivers WHERE car_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteCarIdFromCarsDriversStatement =
+                        connection.prepareStatement(deleteCarIdFromCarsDriversQuery)) {
+            deleteCarIdFromCarsDriversStatement.setLong(1, carId);
+            deleteCarIdFromCarsDriversStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete carId " + carId
+                    + " from cars_drivers table", e);
+        }
+    }
+
+    private List<Car> addAndSetDriversToCar(ResultSet resultSet) throws SQLException {
+        List<Car> cars = new ArrayList<>();
+        while (resultSet.next()) {
+            cars.add(getCarFromResultSet(resultSet));
+        }
+        cars.forEach(c -> c.setDrivers(getDriversForCar(c.getId())));
```

Comment:
you now have a nested connection, which we should avoid as it was said in the common mistakes, you can ask @petro-ushchuk later if he meant something specific by the common logic of these methods, but I think he was generally asking about the possibility of a slight code duplication reduction, IMHO we don't have it, so I'd remove this method back as it was